src/Symfony/Component/Debug/Exception/FlattenException.php

```diff
@@ -33,16 +33,26 @@ class FlattenException
     private $file;
     private $line;
 
+    /**
+     * @deprecated since 4.1, use {@see createFromThrowable()} instead.
+     */
     public static function create(\Exception $exception, $statusCode = null, array $headers = array())
+    {
+        @trigger_error(sprintf('"%s" is deprecated since Symfony 4.1. Use "createFromThrowable()" instead.', __METHOD__), E_USER_DEPRECATED);
```

Comment:
maybe we should not deprecate this method and save us the version bumps?

Reply:
a lot of code in this class will never be reached in case the throwable is an `\Error` (see `instanceof *ExtensionInterface` checks). 
If we really want a flatten error, couldn't it just be a new `FlattenError`? extracting the reusable logic in a trait if any

Reply:
Tbh, given the error handling in place I fear that anyways the result will be more confusing than useful here, perhaps it's just me

Reply:
:+1: for FlattenError

Reply:
I did not expect those version bumps to be a problem, but we can add the depreciation at a later point, I guess. I'm going to remove the deprecation an revert the changes I made to other components.

Reply:
If I added a separate class for errors, I would need to duplicate a lot of code, lose the ability to properly type-hint for FlattenException and I'd still have to do type-switches when working with arbitrary throwables. Sorry, but this is really a bad idea, imho.